docs/examples/couchbase.md

```diff
@@ -0,0 +1,272 @@
+
+## Start Couchbase server using Docker Compose
+
+Couchbase Docker images are published at [Docker Hub](https://hub.docker.com/_/couchbase/).
+
+Use the following Compose file to start Couchbase server:
+
+```
+mycouchbase:
+  image: couchbase/server
+  ports:
+    - 8091:8091
+    - 8092:8092 
+    - 8093:8093 
+    - 11210:11210
+```
+
+The purpose of each port exposed is explained at [Couchbase Developer Portal - Network Configuration](http://developer.couchbase.com/documentation/server/4.1/install/install-ports.html).
+
+The Couchbase Server can be started as:
+
+```
+$ docker-compose up -d
+```
+
+Alternatively, can also be started as:
+
+```
+docker run -d -p 8091-8093:8091:8093 -p 11210:11210 couchbase/server
+```
+
+As you can see, its more convenient to start the container using Compose.
+
+Check the status of container:
+
+```
+docker-compose ps
+           Name                       Command                       State                        Ports            
+-----------------------------------------------------------------------------------------------------------------
+couchbaseserver_mycouchbas   /entrypoint.sh couchbase-s   Up                           11207/tcp,                 
+e_1                          ...                                                       0.0.0.0:11210->11210/tcp,  
+                                                                                       11211/tcp, 18091/tcp,      
+                                                                                       18092/tcp,                 
+                                                                                       0.0.0.0:8091->8091/tcp,    
+                                                                                       0.0.0.0:8092->8092/tcp,    
+                                                                                       0.0.0.0:8093->8093/tcp
+```
+
+Note, all the forwarded ports are shown in the `Ports` column.
+
+Logs can be seen:
+
+```
+docker-compose logs
+Attaching to couchbaseserver_mycouchbase_1
+mycouchbase_1 | Starting Couchbase Server -- Web UI available at http://<ip>:8091
+```
+
+## Configure Couchbase Docker container
+
+We're assuming Coucubase is running on a Docker Machine. IP address of the machine can be found using `docker-machine ip <MACHINE-NAME>`. In our case, the machine's IP address is `192.168.99.100`.
+
+If Docker host is running locally then replace IP address with `localhost`.
+
+### Configure memory for Data and Index service
+
+Data, Query and Index are three different services that can be configured on a Couchbase instance. Each service has different operating needs. For example, Query is CPU-intensive operation and so require a faster processor. Index is disk heavy and so require a faster solid state drive. Data needs to be read/written fast and so require more memory.
```

Comment:
so requires a faster processor. Index is disk heavy and so requires a faster solid state drive. Data needs to be read and written fast and so requires more memory.